ecpy_hqc_legacy/tasks/tasks/util/save_tasks.py

```diff
@@ -333,50 +339,87 @@ def perform(self):
+                else:
                     log = logging.getLogger()
-                    msg = ("In {}, impossible to save arrays exceeding one "
+                    msg = ("In {}, impossible to save arrays exceeding two "
                            "dimension. Save file in HDF5 format.")
                     log.error(msg.format(self.name))
                     self.root.should_stop.set()
 
-        if lengths:
-            if len(lengths) > 1:
+        if shapes_1D:
+            if len(shapes_1D) > 1:
+                log = logging.getLogger()
+                msg = ("In {}, impossible to save simultaneously 1D-arrays of "
+                       "different sizes. Save file in HDF5 format.")
+                log.error(msg.format(self.name))
+                self.root.should_stop.set()
+            else:
+                length = shapes_1D.pop()
+
+        if shapes_2D:
+            if len(shapes_2D) > 1:
                 log = logging.getLogger()
-                msg = ("In {}, impossible to save simultaneously arrays of "
+                msg = ("In {}, impossible to save simultaneously 2D-arrays of "
                        "different sizes. Save file in HDF5 format.")
                 log.error(msg.format(self.name))
                 self.root.should_stop.set()
+            elif shapes_1D:
+                if length == shapes_2D[0]:
+                    shape = shapes_2D.pop()
+                else:
+                    log = logging.getLogger()
+                    msg = ("In {}, 1D-arrays and 2D-arrays could not be "
+                           "broadcast together. Save file in HDF5 format.")
+                    log.error(msg.format(self.name))
+                    self.root.should_stop.set()
             else:
-                length = lengths.pop()
+                shape = shapes_2D.pop()
 
         if not self.array_values:
             new_line = '\t'.join([str(val) for val in values]) + '\n'
             self.file_object.write(new_line.encode('utf-8'))
             self.file_object.flush()
         else:
             columns = []
-            for i, val in enumerate(values):
-                if i in self.array_values:
-                    if val.dtype.names:
-                        columns.extend([val[m] for m in val.dtype.names])
+            if not (2 in self.array_dims):
+                for i, val in enumerate(values):
+                    if i in self.array_values:
+                        if val.dtype.names:
+                            columns.extend([val[m] for m in val.dtype.names])
+                        else:
+                            columns.append(val)
                     else:
-                        columns.append(val)
-                else:
-                    columns.append(numpy.ones(length)*val)
+                        columns.append(numpy.ones(length)*val)
+            else:
+                for i, val in enumerate(values):
+                    print('arghhhhh')
```

Comment:
WTH ?

Reply:
sometimes writing code drives you crazy. Removed.